### PR TITLE
 [PM-8004] [AC-2603] [AC-2616] [AC-2621] [AC-2622] Unmanaged collection fixes

### PR DESCRIPTION
**Pull request by shane-melton**
## Type of change

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Orphaned/Unmanaged collections broke many assumptions in our permission getters and required complicated client side logic to determine if a user has the ability to manage the orphaned collection. We decided to move that logic into the serverside sproc in https://github.com/bitwarden/server/pull/4108 in the form of an `Unmanaged` property on the collection.

This PR utilizes the new `Unmanaged` property in the `CollectionAdminView` permission getters (`canEdit`, `canEditUserAccess`, etc..) to avoid the need to calculate in multiple places. Now that its in the getters, this PR also removes/updates the repeated logic that was implemented for the "AddAccess" badge story.

### `CollectionAdminView`

- Add the `Unmanaged` property
- Remove helpers that were used to determine if a collection was unmanaged client-side.
- Update other helpers that depend on `Unmanaged` to take it into consideration appropriately.

### `org-vault/vault.component.ts`

- Remove the logic that was used to calculate the `AddAccess` property on collections
  -  This is replaced with the new `Unmanaged` property
- Cleanup the `CollectionAdminView` types to make it more clear what's available in the org vault.

### `domain/organization.ts`

- Add helper to determine if the user has access to edit `Unmanaged` collections
  - Limited to admins/owners, providers, and custom users with edit any collection

### `vault-items/vault-items.component.ts`

- Remove now redundant logic for collection permissions and only rely on the permission getters

### `vault-items/vault-collection-row.component.ts`

- Since the logic to determine if `AddAccess` was removed from the `CollectionView` I had to re-add it at the component level. Fortunately, it's able to re-use the existing helpers/flags.

### Service, Models, Responses, everything else

- Add the `Unmanaged` property to all the relevant models/responses

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/web/src/app/vault/core/views/collection-admin.view.ts
Comment (eliykat) [inline apps/web/src/app/vault/core/views/collection-admin.view.ts:1]: Business logic question: can owners/admins also delete unmanaged collections?

**Review by eliykat — APPROVED** [review]
Looks like a great improvement to this logic 👏  just 1 question around delete permissions, otherwise lgtm to me from the permissions side.

File: apps/web/src/app/vault/core/views/collection-admin.view.ts
Comment (shane-melton) [reply apps/web/src/app/vault/core/views/collection-admin.view.ts:1]: I don't believe so based on this copy we added: 
image

Comment (eliykat) [reply apps/web/src/app/vault/core/views/collection-admin.view.ts:1]: Makes sense to me, thanks.

**Review by gbubemismith — APPROVED** [review]
This looks really good and much cleaner. Thanks for working on this @shane-melton

**Review by Jingo88 — APPROVED** [review]

**Comment by shane-melton** [comment]
@Jingo88 @eliykat @gbubemismith 
Apologies for the re-ping, I realized this wasn't hiding the Add Access badge/toggle when the FC V1 flag was disabled and had to tweak that logic.

**Review by gbubemismith — APPROVED** [review]

**Review by gbubemismith — APPROVED** [review]
LGTM

**Review by Jingo88 — APPROVED** [review]

**Review by vincentsalucci — APPROVED** [review]
Approving on behalf of @eliykat

**Pull request merged**